opendatahub-io/odh-dashboard
# Create pipeline version upload modal

JIRA: [RHOAIENG-269](https://issues.redhat.com/browse/RHOAIENG-269)

## Description


Create a modal which allows user to upload a version for a specific pipeline

Implementation:

<img width="1725" alt="Screenshot 2023-12-06 at 3 26 18 PM" src="https://github.com/opendatahub-io/odh-dashboard/assets/37624318/94769b3e-7dc2-4603-8280-cc520ef0ac65">

When the pipeline is selected, it will automatically generate the name for the version, user can edit it, too

<img width="1725" alt="Screenshot 2023-12-06 at 3 27 09 PM" src="https://github.com/opendatahub-io/odh-dashboard/assets/37624318/b3e7dc2f-93bc-480b-aa3d-5947a2e6c2d5">

Added a split button on the global pipelines page, which supports both uploading the pipelines and uploading the pipeline versions

<img width="1725" alt="Screenshot 2023-12-06 at 3 27 27 PM" src="https://github.com/opendatahub-io/odh-dashboard/assets/37624318/0004732c-43ac-49b5-853e-0dbcc60b649e">

Add a dropdown item in the pipeline row action menu, which allows the user to upload new version to a specific

<img width="1725" alt="Screenshot 2023-12-06 at 3 28 26 PM" src="https://github.com/opendatahub-io/odh-dashboard/assets/37624318/d3b6a581-d114-4cae-bddc-2cd57001011b">

On create run page, add 2 link buttons to support uploading pipelines and pipeline versions on this page

<img width="1725" alt="Screenshot 2023-12-06 at 3 31 29 PM" src="https://github.com/opendatahub-io/odh-dashboard/assets/37624318/22940f2c-103e-4293-9c56-fddca6c37cfe">

## How Has This Been Tested?



1. Go to the global pipelines page
2. Try to upload a version or upload a pipeline
3. You could verify where the version is successfully uploaded in the create run page, if you successfully upload it, you should be able to select it when you create a run
4. Try to upload a version in the create run page, it will be auto-selected after you create it
5. Try the upload new version button in the pipeline table row action, it will auto-fill the pipeline selection in the modal

## Test Impact


Added a storybook test for the modal, mainly test whether it could generate the version name based on the change of the pipeline selection.

## Request review criteria:



Self checklist (all need to be checked):

If you have UI changes: 


After the PR is posted & before it merges:
